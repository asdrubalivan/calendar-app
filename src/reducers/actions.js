export const ADD_REMINDER = 'ADD_REMINDER';
export const EDIT_REMINDER = 'EDIT_REMINDER';
export const NEXT_MONTH = 'NEXT_MONTH';
export const PREV_MONTH = 'PREV_MONTH';

export const addReminder = ({ color, reminder, date }) => ({
  date,
  reminder,
  color,
  type: ADD_REMINDER,
});

export const editReminder = (id, { color, reminder, date }) => ({
  id,
  color,
  reminder,
  date,
  type: EDIT_REMINDER,
});

export const nextMonth = () => ({
  type: NEXT_MONTH,
});

export const prevMonth = () => ({
  type: PREV_MONTH,
});
