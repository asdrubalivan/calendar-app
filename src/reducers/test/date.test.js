import reducer from '../calendar';
import { setDate, nextMonth, prevMonth } from '../actions';

const getInitialState = () => reducer(undefined, {});

describe('date reducer', () => {
  const date = '2019-04-01T04:00:00.000Z';
  it('creates a default state', () => {
    const result = getInitialState();
    expect(result).toMatchObject({
      date: null,
      calendarDate: null,
      days: [],
    });
  });
  it('creates an array of days based on the date', () => {
    const initialState = getInitialState();
    const result = reducer(initialState, setDate(date));
    expect(result.date).toBe(date);
    expect(result.calendarDate).toBe(date);
    expect(result.days).toMatchSnapshot('days');
  });
  it('goes to next month', () => {
    const initialState = reducer(getInitialState(), setDate(date));
    const result = reducer(initialState, nextMonth());
    expect(result.date).toBe(date);
    expect(result.calendarDate).toBe('2019-05-01T00:00:00.000Z');
    expect(result.days).toMatchSnapshot('next month 1st');
    const secondMonth = reducer(result, nextMonth());
    expect(secondMonth.date).toBe(date);
    expect(secondMonth.calendarDate).toBe('2019-06-01T00:00:00.000Z');
    expect(secondMonth.days).toMatchSnapshot('next month 2nd');
  });
  it('goes to previous month', () => {
    const initialState = reducer(getInitialState(), setDate(date));
    const result = reducer(initialState, prevMonth());
    expect(result.date).toBe(date);
    expect(result.calendarDate).toBe('2019-03-01T00:00:00.000Z');
    expect(result.days).toMatchSnapshot('prev month 1st');
    const secondMonth = reducer(result, prevMonth());
    expect(secondMonth.date).toBe(date);
    expect(secondMonth.calendarDate).toBe('2019-02-01T00:00:00.000Z');
    expect(secondMonth.days).toMatchSnapshot('prev month 2nd');
  });
});
