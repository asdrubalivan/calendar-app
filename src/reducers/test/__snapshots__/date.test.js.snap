// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date reducer creates an array of days based on the date: days 1`] = `
Array [
  "2019-03-31T00:00:00.000Z",
  "2019-04-01T00:00:00.000Z",
  "2019-04-02T00:00:00.000Z",
  "2019-04-03T00:00:00.000Z",
  "2019-04-04T00:00:00.000Z",
  "2019-04-05T00:00:00.000Z",
  "2019-04-06T00:00:00.000Z",
  "2019-04-07T00:00:00.000Z",
  "2019-04-08T00:00:00.000Z",
  "2019-04-09T00:00:00.000Z",
  "2019-04-10T00:00:00.000Z",
  "2019-04-11T00:00:00.000Z",
  "2019-04-12T00:00:00.000Z",
  "2019-04-13T00:00:00.000Z",
  "2019-04-14T00:00:00.000Z",
  "2019-04-15T00:00:00.000Z",
  "2019-04-16T00:00:00.000Z",
  "2019-04-17T00:00:00.000Z",
  "2019-04-18T00:00:00.000Z",
  "2019-04-19T00:00:00.000Z",
  "2019-04-20T00:00:00.000Z",
  "2019-04-21T00:00:00.000Z",
  "2019-04-22T00:00:00.000Z",
  "2019-04-23T00:00:00.000Z",
  "2019-04-24T00:00:00.000Z",
  "2019-04-25T00:00:00.000Z",
  "2019-04-26T00:00:00.000Z",
  "2019-04-27T00:00:00.000Z",
  "2019-04-28T00:00:00.000Z",
  "2019-04-29T00:00:00.000Z",
  "2019-04-30T00:00:00.000Z",
  "2019-05-01T00:00:00.000Z",
  "2019-05-02T00:00:00.000Z",
  "2019-05-03T00:00:00.000Z",
  "2019-05-04T00:00:00.000Z",
]
`;

exports[`date reducer goes to next month: next month 1st 1`] = `
Array [
  "2019-04-28T00:00:00.000Z",
  "2019-04-29T00:00:00.000Z",
  "2019-04-30T00:00:00.000Z",
  "2019-05-01T00:00:00.000Z",
  "2019-05-02T00:00:00.000Z",
  "2019-05-03T00:00:00.000Z",
  "2019-05-04T00:00:00.000Z",
  "2019-05-05T00:00:00.000Z",
  "2019-05-06T00:00:00.000Z",
  "2019-05-07T00:00:00.000Z",
  "2019-05-08T00:00:00.000Z",
  "2019-05-09T00:00:00.000Z",
  "2019-05-10T00:00:00.000Z",
  "2019-05-11T00:00:00.000Z",
  "2019-05-12T00:00:00.000Z",
  "2019-05-13T00:00:00.000Z",
  "2019-05-14T00:00:00.000Z",
  "2019-05-15T00:00:00.000Z",
  "2019-05-16T00:00:00.000Z",
  "2019-05-17T00:00:00.000Z",
  "2019-05-18T00:00:00.000Z",
  "2019-05-19T00:00:00.000Z",
  "2019-05-20T00:00:00.000Z",
  "2019-05-21T00:00:00.000Z",
  "2019-05-22T00:00:00.000Z",
  "2019-05-23T00:00:00.000Z",
  "2019-05-24T00:00:00.000Z",
  "2019-05-25T00:00:00.000Z",
  "2019-05-26T00:00:00.000Z",
  "2019-05-27T00:00:00.000Z",
  "2019-05-28T00:00:00.000Z",
  "2019-05-29T00:00:00.000Z",
  "2019-05-30T00:00:00.000Z",
  "2019-05-31T00:00:00.000Z",
  "2019-06-01T00:00:00.000Z",
]
`;

exports[`date reducer goes to next month: next month 2nd 1`] = `
Array [
  "2019-05-26T00:00:00.000Z",
  "2019-05-27T00:00:00.000Z",
  "2019-05-28T00:00:00.000Z",
  "2019-05-29T00:00:00.000Z",
  "2019-05-30T00:00:00.000Z",
  "2019-05-31T00:00:00.000Z",
  "2019-06-01T00:00:00.000Z",
  "2019-06-02T00:00:00.000Z",
  "2019-06-03T00:00:00.000Z",
  "2019-06-04T00:00:00.000Z",
  "2019-06-05T00:00:00.000Z",
  "2019-06-06T00:00:00.000Z",
  "2019-06-07T00:00:00.000Z",
  "2019-06-08T00:00:00.000Z",
  "2019-06-09T00:00:00.000Z",
  "2019-06-10T00:00:00.000Z",
  "2019-06-11T00:00:00.000Z",
  "2019-06-12T00:00:00.000Z",
  "2019-06-13T00:00:00.000Z",
  "2019-06-14T00:00:00.000Z",
  "2019-06-15T00:00:00.000Z",
  "2019-06-16T00:00:00.000Z",
  "2019-06-17T00:00:00.000Z",
  "2019-06-18T00:00:00.000Z",
  "2019-06-19T00:00:00.000Z",
  "2019-06-20T00:00:00.000Z",
  "2019-06-21T00:00:00.000Z",
  "2019-06-22T00:00:00.000Z",
  "2019-06-23T00:00:00.000Z",
  "2019-06-24T00:00:00.000Z",
  "2019-06-25T00:00:00.000Z",
  "2019-06-26T00:00:00.000Z",
  "2019-06-27T00:00:00.000Z",
  "2019-06-28T00:00:00.000Z",
  "2019-06-29T00:00:00.000Z",
  "2019-06-30T00:00:00.000Z",
  "2019-07-01T00:00:00.000Z",
  "2019-07-02T00:00:00.000Z",
  "2019-07-03T00:00:00.000Z",
  "2019-07-04T00:00:00.000Z",
  "2019-07-05T00:00:00.000Z",
  "2019-07-06T00:00:00.000Z",
]
`;

exports[`date reducer goes to previous month: prev month 1st 1`] = `
Array [
  "2019-02-24T00:00:00.000Z",
  "2019-02-25T00:00:00.000Z",
  "2019-02-26T00:00:00.000Z",
  "2019-02-27T00:00:00.000Z",
  "2019-02-28T00:00:00.000Z",
  "2019-03-01T00:00:00.000Z",
  "2019-03-02T00:00:00.000Z",
  "2019-03-03T00:00:00.000Z",
  "2019-03-04T00:00:00.000Z",
  "2019-03-05T00:00:00.000Z",
  "2019-03-06T00:00:00.000Z",
  "2019-03-07T00:00:00.000Z",
  "2019-03-08T00:00:00.000Z",
  "2019-03-09T00:00:00.000Z",
  "2019-03-10T00:00:00.000Z",
  "2019-03-11T00:00:00.000Z",
  "2019-03-12T00:00:00.000Z",
  "2019-03-13T00:00:00.000Z",
  "2019-03-14T00:00:00.000Z",
  "2019-03-15T00:00:00.000Z",
  "2019-03-16T00:00:00.000Z",
  "2019-03-17T00:00:00.000Z",
  "2019-03-18T00:00:00.000Z",
  "2019-03-19T00:00:00.000Z",
  "2019-03-20T00:00:00.000Z",
  "2019-03-21T00:00:00.000Z",
  "2019-03-22T00:00:00.000Z",
  "2019-03-23T00:00:00.000Z",
  "2019-03-24T00:00:00.000Z",
  "2019-03-25T00:00:00.000Z",
  "2019-03-26T00:00:00.000Z",
  "2019-03-27T00:00:00.000Z",
  "2019-03-28T00:00:00.000Z",
  "2019-03-29T00:00:00.000Z",
  "2019-03-30T00:00:00.000Z",
  "2019-03-31T00:00:00.000Z",
  "2019-04-01T00:00:00.000Z",
  "2019-04-02T00:00:00.000Z",
  "2019-04-03T00:00:00.000Z",
  "2019-04-04T00:00:00.000Z",
  "2019-04-05T00:00:00.000Z",
  "2019-04-06T00:00:00.000Z",
]
`;

exports[`date reducer goes to previous month: prev month 2nd 1`] = `
Array [
  "2019-01-27T00:00:00.000Z",
  "2019-01-28T00:00:00.000Z",
  "2019-01-29T00:00:00.000Z",
  "2019-01-30T00:00:00.000Z",
  "2019-01-31T00:00:00.000Z",
  "2019-02-01T00:00:00.000Z",
  "2019-02-02T00:00:00.000Z",
  "2019-02-03T00:00:00.000Z",
  "2019-02-04T00:00:00.000Z",
  "2019-02-05T00:00:00.000Z",
  "2019-02-06T00:00:00.000Z",
  "2019-02-07T00:00:00.000Z",
  "2019-02-08T00:00:00.000Z",
  "2019-02-09T00:00:00.000Z",
  "2019-02-10T00:00:00.000Z",
  "2019-02-11T00:00:00.000Z",
  "2019-02-12T00:00:00.000Z",
  "2019-02-13T00:00:00.000Z",
  "2019-02-14T00:00:00.000Z",
  "2019-02-15T00:00:00.000Z",
  "2019-02-16T00:00:00.000Z",
  "2019-02-17T00:00:00.000Z",
  "2019-02-18T00:00:00.000Z",
  "2019-02-19T00:00:00.000Z",
  "2019-02-20T00:00:00.000Z",
  "2019-02-21T00:00:00.000Z",
  "2019-02-22T00:00:00.000Z",
  "2019-02-23T00:00:00.000Z",
  "2019-02-24T00:00:00.000Z",
  "2019-02-25T00:00:00.000Z",
  "2019-02-26T00:00:00.000Z",
  "2019-02-27T00:00:00.000Z",
  "2019-02-28T00:00:00.000Z",
  "2019-03-01T00:00:00.000Z",
  "2019-03-02T00:00:00.000Z",
]
`;
